docs/contributing/README.md

```diff
@@ -45,22 +47,29 @@ mkdocs serve
 
 ## Testing
 
-```bash
-pip install -r requirements/dev.txt
+=== "NVIDIA CUDA"
 
-# Linting, formatting and static type checking
-pre-commit install --hook-type pre-commit --hook-type commit-msg
+    ```bash
+    sudo apt install python3-dev
```

Comment:
This seems like more of a Python setup issue than a vLLM specific issue.

Also, when using `uv`, is it necessary to install a system Python like this anyway? 

Also 2, developers might not have `sudo` access on their dev machines and this may lead them to think they are bocked when they aren't.

Reply:
> This seems like more of a Python setup issue than a vLLM specific issue.

agreed, but it's relevant for getting a working dev env set up from scratch for this repo. Is there a better place to put this? As a note admonition below?

> Also, when using `uv`, is it necessary to install a system Python like this anyway?

I think so. See https://github.com/vllm-project/vllm/pull/18272/#discussion_r2093487251 `` where there's a `fatal error: Python.h: No such file or directory` without this.

Reply:
Yeah an admonition would be better, it's more of a troubleshooting tip than a normal step for setting up your dev environment

Reply:
done!